src/store/breakoutRoomsStore.js

```diff
@@ -32,68 +32,81 @@ import {
 	reorganizeAttendees,
 } from '../services/breakoutRoomsService.js'
 import { showError } from '@nextcloud/dialogs'
-import { set } from 'vue'
+import Vue from 'vue'
 import { emit } from '@nextcloud/event-bus'
 
 const state = {
-	breakoutRoomsReferences: {},
+	breakoutRooms: {},
 }
 
 const getters = {
-	breakoutRoomsReferences: (state) => (token) => {
-		if (!state.breakoutRoomsReferences?.[token]) {
+	breakoutRooms: (state) => (token) => {
+		if (!state.breakoutRooms?.[token]) {
 			return []
 		}
-		return state.breakoutRoomsReferences?.[token]
-	},
-
-	hasBreakoutRooms: (state, getters, rootState) => (token) => {
-		if (!state.breakoutRoomsReferences?.[token]) {
-			return false
-		}
-		return !!state.breakoutRoomsReferences?.[token]
-			.every(breakoutRoomToken => rootState.conversationsStore.conversations?.[breakoutRoomToken])
+		return state.breakoutRooms?.[token]
 	},
 }
 
 const mutations = {
-	addBreakoutRoomsReferences(state, { token, breakoutRoomsReferences }) {
-		if (!state.breakoutRoomsReferences[token]) {
-			set(state.breakoutRoomsReferences, token, [])
+	/**
+	 * Adds a breakout room to the store.
+	 *
+	 * @param {object} state current store state;
+	 * @param {object} conversation the conversation;
+	 * @param conversation.parentRoomToken
+	 * @param parentRoomToken the parent room token;
+	 * @param breakoutRoom
+	 * @param conversation.breakoutRoom
+	 */
+	addBreakoutRoom(state, { parentRoomToken, breakoutRoom }) {
+		if (!state.breakoutRooms[parentRoomToken]) {
+			Vue.set(state.breakoutRooms, parentRoomToken, [])
 		}
-		state.breakoutRoomsReferences[token] = breakoutRoomsReferences
+		Vue.set(state.breakoutRooms, parentRoomToken, [...state.breakoutRooms[parentRoomToken], breakoutRoom])
 	},
 
-	removeReferences(state, token) {
-		set(state.breakoutRoomsReferences, token, [])
+	deleteBreakoutRooms(state, token) {
+		Vue.delete(state.breakoutRooms, token)
 	},
 }
 
```

Comment:
And add separate method for deleting only one room?
```suggestion
deleteBreakoutRoom(state, parentRoomToken, token) {
		Vue.set(state.breakoutRooms, parentRoomToken, state.breakoutRooms[parentRoomToken]
		.filter(room => room[token] !== token))
	},
}
```

Reply:
In the next pr where we add this feature :)